hal-x86_64/src/interrupt/pic.rs

```diff
@@ -48,9 +48,10 @@ impl CascadedPic {
 }
 
 impl hal_core::interrupt::Control for CascadedPic {
+    type Registers = super::Registers;
     fn register_handlers<H>(&mut self) -> Result<(), hal_core::interrupt::RegistrationError>
     where
-        H: Handlers,
+        H: Handlers<super::Registers>,
```

Comment:
Kinda hate that this forces the PICs to pretend to be aware of CPU exception stack frames...but I think the whole interrupt handling interface that currently exists will need a bigger refactor before it makes sense for paripheral-driven interrupts (PIC should probably implement a different trait...)

Reply:
That seems like a reasonable plan to me!